### PR TITLE
BrAPI setting enhancements

### LINKED ISSUE #570: Hide BrAPI in import/export dialogs if it hasn't been set up
- Potentially just add a setting at the top of the brapi settings section to 'Enable BrAPI'

### PR DESCRIPTION
**Pull request by bellerbrock**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

Adds checkbox to enable/disable BrAPI. Hides remaining prefs and removes BrAPI from import/export options if disabled.
Adds editable server display name to use in import/export dialogs instead of 'BrAPI'. Defaults to the server url minus scheme and subdomain.
Gives user the option to make BrAPI the default import and/or export option when a server is successfully authorized.

closes #570 and #699

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
BrAPI setting enhancements
```

### CODE REVIEW

**Comment by trife** [comment]
@bellerbrock add a check to the server name so that if it's blank it shows "BrAPI". Are there checks for if servername returns blank?

File: app/src/main/java/com/fieldbook/tracker/preferences/BrapiPreferencesFragment.java [hunk 1]
```diff
@@ -111,10 +145,12 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
         }
 
         //set saved urls, default to the test server
-        String url = prefMgr.getSharedPreferences().getString(GeneralKeys.BRAPI_BASE_URL, getString(R.string.brapi_base_url_default));
-        String oidcUrl = prefMgr.getSharedPreferences().getString(GeneralKeys.BRAPI_OIDC_URL, getString(R.string.brapi_oidc_url_default));
+        String url = sp.getString(GeneralKeys.BRAPI_BASE_URL, getString(R.string.brapi_base_url_default));
+        String displayName = sp.getString(GeneralKeys.BRAPI_DISPLAY_NAME, getString(R.string.preferences_brapi_server_test));
```
Comment (chaneylc) [inline app/src/main/java/com/fieldbook/tracker/preferences/BrapiPreferencesFragment.java:149]: In TraitEditor and Config activities you do this:
`String displayName = ep.getString(GeneralKeys.BRAPI_DISPLAY_NAME, "BrAPI");`

It seems this should be consistent and all cases should be R.string.export_source_brapi, but let me know if you did this for a specific reason.

**Review by chaneylc — CHANGES_REQUESTED** [review]
- [x] disable toolbar authentication if BrAPI is disabled

File: app/src/main/java/com/fieldbook/tracker/preferences/BrapiPreferencesFragment.java (hunk 1 above)
Comment (bellerbrock) [reply app/src/main/java/com/fieldbook/tracker/preferences/BrapiPreferencesFragment.java:149]: The idea was that when BrAPI is first enabled the url defaults to the test server, so the display name should default to the test server text. From that point forward the BRAPI_BASE_URL is always set to something, so the default can b anything, it no longer gets used. But regardless you're right it should be consistent if only for readability, I'll go ahead and unify them.

Speaking of the default display name, how do you feel about changing the `preferences_brapi_server_test` string from '**Default Test Server**' to '**BrAPI Test Server**'? I think that could help clarify things.

Comment (trife) [reply app/src/main/java/com/fieldbook/tracker/preferences/BrapiPreferencesFragment.java:149]: Yeah that's fine with me

**Review by trife — APPROVED** [review]

**Pull request merged**